nidirect_taxonomy_navigator/src/TaxonomyAdminBreadcrumb.php

```diff
@@ -0,0 +1,161 @@
+ * using the taxonomy_navigator module.
+ *
+ * Extends the core path based breadcrumb builder but slightly
+ * adjusts how the trail is assembled; see build() below.
+ *
+ * @package Drupal\nidirect_taxonomy_navigator
+ */
+class TaxonomyAdminBreadcrumb extends PathBasedBreadcrumbBuilder implements BreadcrumbBuilderInterface {
+
+  /**
+   * The router request context.
+   *
+   * @var \Drupal\Core\Routing\RequestContext
+   */
+  protected $context;
+
+  /**
+   * The menu link access service.
+   *
+   * @var \Drupal\Core\Access\AccessManagerInterface
+   */
+  protected $accessManager;
+
+  /**
+   * Route matches from the service container parameters.
+   *
+   * @var array
+   */
+  protected $routeMatches;
+
+  /**
+   * Request stack service.
+   *
+   * @var \Symfony\Component\HttpFoundation\RequestStack
+   */
+  protected $requestStack;
+
+  /**
+   * Constructs the TaxonomyAdminBreadcrumb.
+   *
+   * @param \Drupal\Core\Routing\RequestContext $context
+   *   The router request context.
+   * @param \Drupal\Core\Access\AccessManagerInterface $access_manager
+   *   The menu link access service.
+   * @param \Symfony\Component\Routing\Matcher\RequestMatcherInterface $router
+   *   The dynamic router service.
+   * @param \Drupal\Core\PathProcessor\InboundPathProcessorInterface $path_processor
+   *   The inbound path processor.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The config factory service.
+   * @param \Drupal\Core\Controller\TitleResolverInterface $title_resolver
+   *   The title resolver service.
+   * @param \Drupal\Core\Session\AccountInterface $current_user
+   *   The current user object.
+   * @param \Drupal\Core\Path\CurrentPathStack $current_path
+   *   The current path.
+   * @param \Drupal\Core\Path\PathMatcherInterface $path_matcher
+   *   The path matcher service.
+   * @param array $route_matches
+   *   Matching routes injected from service parameters.
+   */
+  public function __construct(RequestContext $context, AccessManagerInterface $access_manager, RequestMatcherInterface $router, InboundPathProcessorInterface $path_processor, ConfigFactoryInterface $config_factory, TitleResolverInterface $title_resolver, AccountInterface $current_user, CurrentPathStack $current_path, PathMatcherInterface $path_matcher = NULL, array $route_matches) {
+    parent::__construct($context, $access_manager, $router, $path_processor, $config_factory, $title_resolver, $current_user, $current_path, $path_matcher);
+    $this->routeMatches = $route_matches;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function applies(RouteMatchInterface $route_match) {
+    $match = FALSE;
+
+    // Check for matching route names.
+    if (in_array($route_match->getRouteName(), $this->routeMatches)) {
+      $match = TRUE;
+    }
+
+    return $match;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build(RouteMatchInterface $route_match) {
```

Comment:
A good chunk of this is duplicated from Core's fallback class... because we're doing more or less the exact same thing. Key differences are:

- In `applies()` we're checking the route against the service parameters, not just kicking in by default.
- In `build()` we're tweaking the path reference along the likes of `s/taxonomy\/manage/taxonomy_navigator/g`, and letting the remaining access checks and link generation happen as it should. It would have been lovely to be able to alter the path components before passing this back to the parent class for building, but because there's no 'hook' or event here to do that we end up having to take most of the original implementation and include it as part of the overridden method. Not as DRY as I'd like, but necessary.